HotCommands/Commands/MoveMemberUp.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Text.Editor;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace HotCommands
+{
+    /// <summary>
+    /// Command handler for ToggleComment
+    /// </summary>
+    internal sealed class MoveMemberUp
+    {
+        /// <summary>
+        /// VS Package that provides this command, not null.
+        /// </summary>
+        private readonly Package package;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MoveMethodUp"/> class.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        private MoveMemberUp(Package package)
+        {
+            if (package == null)
+            {
+                throw new ArgumentNullException("package");
+            }
+
+            this.package = package;
+        }
+
+        /// <summary>
+        /// Gets the instance of the command.
+        /// </summary>
+        public static MoveMemberUp Instance
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Gets the service provider from the owner package.
+        /// </summary>
+        private IServiceProvider ServiceProvider
+        {
+            get
+            {
+                return this.package;
+            }
+        }
+
+        /// <summary>
+        /// Initializes the singleton instance of the command.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        public static void Initialize(Package package)
+        {
+            Instance = new MoveMemberUp(package);
+        }
+
+        public int HandleCommand(IWpfTextView textView)
+        {
+            var memberslist = new[] { "MethodDeclaration", "PropertyDeclaration", "FieldDeclaration" };
+            MemberDeclarationSyntax prevMember = null;
+            MemberDeclarationSyntax member = null;
+            var cursorPosition = textView.Caret.ContainingTextViewLine.Extent.Span.Start;
+            var members = CSharpSyntaxTree.ParseText(textView.TextSnapshot.GetText()).GetRoot().DescendantNodes().OfType<MemberDeclarationSyntax>();
+
+            for (int i = 0; i < members.Count(); i++)
```

Comment:
A for loop with `Count()` is a bad idea. This is likely already O(n2) because `Count()` will iterate all of the elements of `members` and that will happen on every element of the loop.